src/main/webapp/recipe_post.jsp

```diff
@@ -0,0 +1,95 @@
+                             <input class="form-check-input" type="radio" name="difficulty" id="medium" value="medium" required>
+                             Medium</label>
+                         </div>
+                         <div class="form-check">
+                             <label class="radio-inline" for="hard">
+                             <input class="form-check-input" type="radio" name="difficulty" id="hard" value="hard" required>
+                             Hard</label>
+                         </div>
+             </fieldset>
+             <div class="form-group">
+                 <label>Select the prearation time for your recipe</label>
+                 <div class="form-row">
+                      <label class="form-inline" for="hour">
+                           <select class="form-control m-3" id="hour" name="hour">
+                             <option value="1">1</option>
+                             <option value="2">2</option>
+                             <option value="3">3</option>
+                             <option value="4">more than 3</option>
+                           </select>hour
+                      </label>
+                      <label class="form-inline" for="min">
+                           <select class="form-control m-3" id="min" name="min">
+                             <option value="15">15</option>
+                             <option value="30">30</option>
+                             <option value="45">45</option>
+                           </select>minutes
+                      </label>
+                 </div>
+             </div>
+             <div class="form-group">
+                <label>Ingredients</label>
+                <div class="ingredient-wrapper">
+                <div class="ingredient-form">
+                 <div class="form-row">
+                  <div class="col">
+                      <input type="number" class="form-control" name="ingredients[][quantity]" placeholder="quantity" required>
+                  </div>
+                  <div class="col">
+                      <input type="text" class="form-control" name="ingredients[][measure]" maxlength="50" placeholder="unit of measurment" >
+                  </div>
+                  <div class="col">
+                      <input type="text" class="form-control" name="ingredients[][ingredient]" placeholder="ingredient" maxlength="50" required>
+                  </div>
+                 </div>
+                </div>
+                <a href="javascript:void(0);" class="ingredient-add-button" title="add-step">ADD INGREDIENT</a>
+                </div>
+             </div>
+             <div class="form-group">
+                <label for="step">Steps</label>
+                <div class="step-wrapper">
+                  <div class="step-form">
+                   <textarea class="form-control" name="step[]" rows="3" maxlength="500" required></textarea>
+                  </div>
```

Comment:
What happened with the idea of adding the remove button here and hiding it with CSS instead of adding it with JS?

I think it would be better! Please ping me on chat over this.